### PR TITLE
Update Boost example

### LINKED ISSUE #501: right now it's not possible to add boost properly using CPM
I was disappointed when I saw a not working example with boost. I also was disappointed by the idea of using boost-cmake . 

After sometime reading CMake code of boost I finally got it. it was ridiculous to find out that CPM...package() cannot add "option" with multiple values.

Then, It was interesting to find, that boost cmake code is actually good.

here's the working example:
```
set(TRY_BOOST_VERSION "1.83.0")
set(BOOST_NOT_HEADER_ONLY_COMPONENTS_THAT_YOU_NEED "thread")
set(BOOST_HEADER_ONLY_COMPONENTS_THAT_YOU_NEED "asio")

set(IS_BOOST_LOCAL OFF)
if(${CPM_LOCAL_PACKAGES_ONLY})
  message(STATUS "Trying to find Boost...")
  find_package(Boost ${TRY_BOOST_VERSION} REQUIRED COMPONENTS
    ${BOOST_NOT_HEADER_ONLY_COMPONENTS_THAT_YOU_NEED})
  set(IS_BOOST_LOCAL ON)
elseif(${CPM_USE_LOCAL_PACKAGES} OR NOT ${CPM_DOWNLOAD_ALL})
  message(STATUS "Trying to use local Boost...")
  find_package(Boost ${TRY_BOOST_VERSION} COMPONENTS ${BOOST_NOT_HEADER_ONLY_COMPONENTS_THAT_YOU_NEED})
  if(${BOOST_FOUND})
    set(IS_BOOST_LOCAL ON)
    message(DEBUG "boost include dir: ${Boost_INCLUDE_DIRS}")
  endif()
endif()

if(NOT (${BOOST_FOUND}) OR (NOT DEFINED BOOST_FOUND))
  message(STATUS "Trying to download Boost...")
  set(BOOST_INCLUDE_LIBRARIES
    "${BOOST_NOT_HEADER_ONLY_COMPONENTS_THAT_YOU_NEED};${BOOST_HEADER_ONLY_COMPONENTS_THAT_YOU_NEED}")
  CPMAddPackage(
    NAME Boost
    URL
    "https://github.com/boostorg/boost/releases/download/boost-${TRY_BOOST_VERSION}/boost-${TRY_BOOST_VERSION}.tar.xz"
  )
  set(IS_BOOST_LOCAL OFF)

endif()


if(${IS_BOOST_LOCAL})
  target_link_libraries(${PROJECT_NAME} PUBLIC Boost::boost)
else()
  target_link_libraries(${PROJECT_NAME} PUBLIC Boost::asio)
endif()
target_link_libraries(${PROJECT_NAME} PUBLIC Boost::thread)

```

### PR DESCRIPTION
**Pull request by ScottBailey**
Boost is a great library and it's important to many projects. This PR is written to show an example of escaping input in the OPTIONS argument of `CPMAddPackage()` and converts from git clone to acquiring the source release archive and testing against a hash.

While a git clone of a github repo often makes sense, even a shallow copy of Boost is time consuming due to the many submodules involved. For my 50mbit connection, it takes almost 3 minutes to clone the repo. Using the github releasse cuts the time to 33 seconds.

This is also the only example of getting the tar of a release from github which, I think, should be the most common approach as it requires the least bandwidth.

I believe this may close #501 

Tested in Debian GNU/Linux and Windows 11.

Added information on locating GitHub repository's source archives and determining SHA256 hash values.

### CODE REVIEW

**Review by TheLartians — APPROVED** [review]
This is great, thanks for updating the example to show how to use releases!

Would it make sense to also update the boost example in the readme and explaining the advantage of using a release directly for large projects there?

**Comment by ScottBailey** [comment]
> Would it make sense to also update the boost example in the readme and explaining the advantage of using a release directly for large projects there?

It would make a TON of sense to do that. I had forgotten the example was even in the readme. I'll update that in the next couple days and and ask for a rereview.

**Comment by TheLartians** [comment]
> I'll update that in the next couple days and and ask for a rereview.

@ScottBailey any ETA on this? Otherwise we can also go ahead with merging the examples now and update the Readme in a future PR.

**Comment by ScottBailey** [comment]
> > I'll update that in the next couple days and and ask for a rereview.
> 
> @ScottBailey any ETA on this? Otherwise we can also go ahead with merging the examples now and update the Readme in a future PR.

Let's hold off until Friday COB. I am hoping to have some "free" time then. I made a number of edits but then got pulled away for my paying gig.

File: README.md [hunk 1]
```diff
@@ -475,3 +483,85 @@ For a full example on using CPM to download and configure lua with sol2 see [her
...
+`https://github.com/gabime/spdlog/archive/refs/tags/v1.8.5.tar.gz`
+
+Exactly like the release.
+
+
+#### Commit
+
+If a specific commit contains the code you need, it's defined as follows:
+
+`https://github.com/<user>/<name>/arcive/<commit-hash>.<archive-type>`
```
Comment (TheLartians) [inline README.md:556]: Small typo

```suggestion
`https://github.com/<user>/<name>/archive/<commit-hash>.<archive-type>`
```

**Review by TheLartians — APPROVED** [review]

**Pull request merged**